### PR TITLE
Astro 302 - Taskflow Example Update

### PR DESCRIPTION
**Pull request by im-cr**
**TASK**
- Add associated animated gifs to each page via the new Lightbox implementation
- Use uploaded video with animated gifs as fallback in a <video> element

**IMPACTED AREAS**

**GRM Service UX Design**
- grm-service-ux-design/grm-dashboard/
- grm-service-ux-design/grm-equipment-manager/
- grm-service-ux-design/grm-schedule/

**TT&C Service UX Design**
- ttc-service-ux-design/about-the-ttc-designs/
- ttc-service-ux-design/ttc-monitor/
- ttc-service-ux-design/ttc-command/
- ttc-service-ux-design/ttc-investigate/

### CODE REVIEW

File: 404.md [hunk 1]
```diff
@@ -2,7 +2,6 @@
 layout: interior.template.njk
 title: Oops! Not Found
 permalink: 404.html
-eleventyExcludeFromCollections: true
```
Comment (sknep) [inline 404.md:5]: can i ask why these lines got removed? did you branch from somewhere else? these are used to exclude pages from sitemap generation

File: _content/404.md [hunk 2]
```diff
@@ -2,7 +2,6 @@
 layout: interior.template.njk
 title: Oops! Not Found
 permalink: 404.html
-eleventyExcludeFromCollections: true
```
Comment (sknep) [inline _content/404.md:5]: I think this needs to stay

File: _content/_data/meta.json [hunk 3]
```diff
@@ -1,3 +1,3 @@
 {
-	"version": "4.6"
```
Comment (sknep) [inline _content/_data/meta.json:2]: this is going back in time -- possibly started from an old branch.

File: _content/_data/navigation.json [hunk 4]
```diff
@@ -264,25 +261,6 @@
...
-			{
-				"label": "Propose a Change",
-				"url": "/community/propose-a-change"
-			},
-			{
-				"label": "Content Policy",
-				"url": "/community/content-policy"
-			}
-		]
-	},
```
Comment (sknep) [inline _content/_data/navigation.json:285]: I think it's important that we lose none of these changes.

File: _content/community/content-policy.md [hunk 5]
```diff
@@ -1,87 +0,0 @@
----
```
Comment (sknep) [inline _content/community/content-policy.md:1]: This needs to come back

File: _content/grm-service-ux-design/grm-dashboard.md [hunk 6]
```diff
@@ -83,7 +83,14 @@ If operators choose to drill into an alert via the Investigate button in the Ale
 
 Below is an animated walkthrough of a representative task flow using the GRM Dashboard app. In this flow, the operator notices a critical alert for an upcoming contact, investigates it via the Contact Alert Details page and then resolves the issue by selecting an alternative piece of equipment to use for the contact.
 
-![Task Flow Example](/img/service-specific-ux-design/gif-placeholder.png)
+<div markdown="1">
```
[the remark this answers is not in the document]
Comment (sknep) [reply _content/grm-service-ux-design/grm-dashboard.md:86]: It works, though a custom shortcode in 11ty might be cleaner.

File: _includes/components.template.njk [hunk 7]
```diff
@@ -32,7 +32,7 @@
 					</a>
 				{% endif %}
 				{% if git %}
-					<a class="sample-links__bitbucket" href="https://github.com/RocketCommunicationsInc/astro-componentssrc/components/{{git}}">Bitbucket</a>
```
Comment (sknep) [inline _includes/components.template.njk:35]: Nope, we want github links, not bitbucket ones

File: css/legacy/chapters_legacy.css [hunk 8]
```diff
@@ -0,0 +1,181 @@
+main {
```
Comment (sknep) [inline css/legacy/chapters_legacy.css:1]: Are these files necessary?

File: netlify.toml [hunk 9]
```diff
@@ -68,10 +68,4 @@
 	from = "/tt-c-service-ux-design/tt-c-investigate"
 	to = "/ttc-service-ux-design/ttc-investigate"
 	status = 300
-	force = true
-
```
Comment (sknep) [inline netlify.toml:72]: We still need this, we renamed the component

File: package.json [hunk 10]
```diff
@@ -9,7 +9,7 @@
 	"main": "index.js",
 	"scripts": {
 		"test": "echo \"Error: no test specified\" && exit 1",
-		"sass": "node-sass -rw scss -o css --output-style compressed",
```
Comment (sknep) [inline package.json:12]: AH, so this is why the css files are un-minified in this PR! Is there a reason? if not, they're adding a ton of noise to your PR

File: _includes/base.template.njk [hunk 11]
```diff
@@ -44,5 +45,7 @@
 				{% include "_footer.njk" %}
 			</div>
 		</div>
+
+		<script src="/js/lightbox.js"></script>
```
Comment (sknep) [inline _includes/base.template.njk:49]: This loads the javascript for the lightbox on all pages -- is it used on _every_ page? If not, you could move it (and the stylesheet link) to just the template(s) where it shows up, keeping the site a little faster on non-lightboxed pages (compliance, home, etc)

**Comment by dmcalester** [comment]
Hey Cortney, 

- I think most of @sknep’s comments are a result of this PR request to draft, it should be to master as these changes are fixes that should be up before draft is approved. But please confirm changing the target branch fixes all comments.
- The poster frame for the images should be a static poster image with a play button/icon (see https://draft--astrouxds.netlify.app/fd-service-ux-design/about-the-fd-designs/) for an example
- Sorry I did not make it clear, but the URLs for all MP4/WebM videos should reference the AWS links posted above

**Comment by sknep** [comment]
This is entirely optional, but you might be interested in a CSS-only lightbox implementation. There are a few ways to do this, here's one: https://codepen.io/gschier/pen/HCoqh 

The implementations I've seen use in-page anchor links and `:target` to handle styling the image as, typically, position: absolute, with the scrim (semitransparent black background) on a pseudo-element of the container. When a colleague first showed me this, I had serious doubts, but it surprised me, and I wanted to pass along the neat idea in case you had an interest.

I agree with Duncan, there really needs to be some sort of play button or call to action over or inside the movie's poster image (which usually shows up if someone is on mobile, because videos on mobile don't usually autoplay). Not sure if that was provided to you, as the link Duncan sent above doesn't show a working video :(

**Comment by im-cr** [comment]
@sknep good call on the css lightbox. Added that in with a small tweak around sizing of the lightbox content. Thanks. @dmcalester switched over to master from draft. Updated video paths. 

Also added in 'play' icon over existing placeholder gifs. Do we want to use a static png rather than animated gif placeholder? If not, I can add in small hover effect to better illustrate interaction. Let me know what you both think.

File: _content/grm-service-ux-design/grm-dashboard.md [hunk 12]
```diff
@@ -83,7 +83,21 @@ If operators choose to drill into an alert via the Investigate button in the Ale
...
+	<figure>
+		<a href="#demo">
+			<img src="/img/service-specific-ux-design/grm-dashboard-modify-string-placeholder.gif" 
+			alt="GRM Dashboard demo" />
+		</a>
+	</figure>
+	<a href="#" class="lightbox" id="demo" markdown="1">
```
Comment (sknep) [inline _content/grm-service-ux-design/grm-dashboard.md:93]: When I close the lightbox (by clicking the scrim), it shoots me back up to the top of the page in chrome. Maybe call this `#close` so that it doesn't jump the page when you exit the lightbox?

File: _content/grm-service-ux-design/grm-dashboard.md [hunk 13]
```diff
@@ -83,7 +83,21 @@ If operators choose to drill into an alert via the Investigate button in the Ale
...
+		<video width="99%" autoplay loop muted markdown="1" controls
```
Comment (sknep) [inline _content/grm-service-ux-design/grm-dashboard.md:94]: I think the controls look a little weird on this, since its already autoplaying and looping. Did the designer really want the control bar?

File: scss/lightbox.scss [hunk 14]
```diff
@@ -0,0 +1,392 @@
+// .lb-loader,.lightbox{
```
Comment (sknep) [inline scss/lightbox.scss:1]: That's a lot to comment out -- if we don't need it, feel free to remove

File: scss/lightbox.scss [hunk 15]
```diff
@@ -0,0 +1,392 @@
...
+//   transition: opacity 0.2s;
+// }
+
+// .lb-data .lb-close:hover {
+//   cursor: pointer;
+//   filter: progid:DXImageTransform.Microsoft.Alpha(Opacity=100);
+//   opacity: 1;
+// }
+
+.lightbox{
```
Comment (sknep) [inline scss/lightbox.scss:370]: I love it! so short and sweet

**Comment by sknep** [comment]
> Also added in 'play' icon over existing placeholder gifs. Do we want to use a static png rather than animated gif placeholder? If not, I can add in small hover effect to better illustrate interaction. Let me know what you both think.

Something with a play button probably shouldn't be _moving_ underneath it. I think static png is better, personally. But was the original ask to implement both gif and mp4? I'm wondering why the designer gave you both assets. 

Hover effect sounds shiny! I'm all for seeing it :)

**Comment by sknep** [comment]
Grr, the iFramed sales/support chat can sit on top of the video at some sizes. Probably not worth fixing but it is annoying:
Screen Shot 2020-07-28 at 5 22 56 PM

File: _content/grm-service-ux-design/grm-dashboard.md [hunk 16]
```diff
@@ -83,7 +83,21 @@ If operators choose to drill into an alert via the Investigate button in the Ale
...
+		poster="/img/service-specific-ux-design/grm-dashboard-modify-string.gif">
+			<source src="http://com.rocketcom.astrouxds.s3.us-west-2.amazonaws.com/media/grm-dashboard-modify-string.mp4" type="video/mp4" markdown="1">
```
Comment (sknep) [inline _content/grm-service-ux-design/grm-dashboard.md:96]: These all need to be at https, to avoid a security console error

Comment (sknep) [reply _content/grm-service-ux-design/grm-dashboard.md:96]: It looks like we have an SSL error here too:
`GET https://com.rocketcom.astrouxds.s3.us-west-2.amazonaws.com/media/grm-equipment-manager-sched-job.webm net::ERR_CERT_COMMON_NAME_INVALID`

File: _content/grm-service-ux-design/grm-dashboard.md (hunk 12 above)
Comment (dmcalester) [reply _content/grm-service-ux-design/grm-dashboard.md:93]: Static image placeholder

File: _content/grm-service-ux-design/grm-dashboard.md (hunk 16 above)
Comment (dmcalester) [reply _content/grm-service-ux-design/grm-dashboard.md:96]: @im-cr @sknep getting the S3 bucket an SSL in the immediate future is going to be tough. There are some entanglements that make it a non-trivial lift. 

For now let’s revert to the animated gif walkthroughs and remove the videos. I’ll work on getting either the SSL situation sorted out or an alternate option.

**Comment by dmcalester** [comment]
Minor rendering bug in Firefox 78.0.2 Play icon appears below the target image covering source materials
Screen Shot 2020-08-07 at 12 54 03 PM

**Review by dmcalester — APPROVED** [review]

**Review by sknep — APPROVED** [review]

**Comment by sknep** [comment]
There's something up with the branches on this repo -- 154 commits in this PR, some from Jeff in February? If this is ready to ship, I won't hold it up, but maybe this repo would benefit from a branch and merge strategy that keeps things cleaner.
